app/api/v1/endpoints/recommended_challenge.py

```diff
@@ -0,0 +1,115 @@
+# routers/recommended_challenge.py
+
+from fastapi import APIRouter, Depends, HTTPException, status
+from sqlalchemy.ext.asyncio import AsyncSession
+from uuid import UUID
+from typing import List
+
+from app.db.session import get_db
+from app.enum.category import CategoryNameEnum
+
+from app.schemas.base import ResponseBase
+from app.schemas.recommended_challenge import (
+    RecommendedChallengeCreate,
+    RecommendedChallengeUpdate,
+    RecommendedChallengeResponse,
+)
+from app.services.recommended_challenge import (
+    select_recommended_challenge_by_id,
+    select_recommended_challenges,
+    register_recommended_challenge,
+    update_recommended_challenge_by_id,
+    delete_recommended_challenge_by_id,
+)
+
+
+router = APIRouter()
+
+
+# 전체 조회 (user_id + category_name)
+@router.get(
+    "/user/{user_id}/category/{category_name}",
+    response_model=ResponseBase[List[RecommendedChallengeResponse]],
+)
+async def get_recommended_challenges(
+    user_id: UUID,
+    category_name: str,
```

Comment:
```suggestion
    category_id: UUID,
```
여기도 이제 위처럼 수정 되면 될 거 같습니다.